docs/resources/aws_s3_bucket.md

```diff
@@ -0,0 +1,123 @@
+
+The value of bucket_acl is an Array of simple objects.  Each object has a `permission` property and a `grantee` property.  The `permission` property will be a string such as 'READ', 'WRITE' etc (See the [AWS documentation](https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#get_bucket_acl-instance_method) for a full list).  The `grantee` property contains sub-properties, such as `type` and `uri`.
+
+    
+    bucket_acl = aws_s3_bucket('my-bucket')
+
+    # Look for grants to "AllUsers" (that is, the public)
+    all_users_grants = bucket_acl.select do |g|
+      g.grantee.type == 'Group' && g.grantee.uri =~ /AllUsers/
+    end
+
+    # Look for grants to "AuthenticatedUsers" (that is, any authenticated AWS user - nearly public)
+    auth_grants = bucket_acl.select do |g|
+      g.grantee.type == 'Group' && g.grantee.uri =~ /AuthenticatedUsers/
+    end
+
+### bucket_policy
+
+The `bucket_policy` is a low-level property that describes the IAM policy document controlling access to the bucket.  Other higher-level properties, such as be\_public, are more concise and easier to use.  The `bucket_policy` property returns a Ruby structure that you can probe to check for particular statements.
+
+The `bucket_policy` property returns an Array of simple objects, each object being an IAM Policy Statement. See the [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html#example-bucket-policies-use-case-2) for details about the structure of this data.
+
+If there is no bucket policy, this property will return an empty Array.
+
+    bucket_policy = aws_s3_bucket('my-bucket')
+
+    # Look for statements that allow the general public to do things
+    # This may be a false positive;  it's possible these statements
+    # could be protected by conditions, such as IP restrictions.
+    public_statements = bucket_policy.select do |s|
+      s.effect == 'Allow' && s.principal == '*'
+    end
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers (such as `exist`) please visit our [matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+### be_public
+
+The `be_public` matcher tests if the bucket has access controls that are potentially insecure.  The intent is that this high-level matcher may be used to detect a number of insecure conditions, which may be added to in the future.  The matcher currently will report an insecure bucket if _any_ of the following conditions are met:
+
+  1. A bucket ACL grant exists for the 'AllUsers' group
+  2. A bucket ACL grant exists for the 'AuthenticatedUsers' group
+  3. A bucket policy has an effect 'Allow' and principal '*'
+
+Note in particular that this resource currently does not detect insecure object ACLs.
```

Comment:
Note: This resource does not detect insecure object ACLs.

Reply:
agree